companyweb_base/models/partner_cweb_model.py

```diff
@@ -0,0 +1,527 @@
+        cweb_has_companystatus_value = cweb_response["CompanyStatus"]["Value"]
+        if self.cweb_companystatus_enable and cweb_has_companystatus_value:
+            self.cweb_companystatus = cweb_response["CompanyStatus"]["Value"]["Info"]
+        elif self.cweb_companystatus_enable and not cweb_has_companystatus_value:
+            self.cweb_companystatus_enable = False
+        else:
+            self.cweb_companystatus = None
+
+        self.cweb_vat_liable_enable = cweb_response["VatEnabled"]["IsEnabled"]
+        cweb_has_vat_liable_value = cweb_response["VatEnabled"]["Value"]
+        if self.cweb_vat_liable_enable and cweb_has_vat_liable_value:
+            self.cweb_vat_liable = cweb_response["VatEnabled"]["Value"]
+        elif self.cweb_vat_liable_enable and not cweb_has_vat_liable_value:
+            self.cweb_vat_liable_enable = False
+        else:
+            self.cweb_vat_liable = None
+
+    def _populate_address_cweb(self, cweb_response):
+        self.cweb_address_enable = cweb_response["Address"]["IsEnabled"]
+        cweb_has_address_values = cweb_response["Address"]["Value"]
+        if self.cweb_address_enable and cweb_has_address_values:
+            self.cweb_street = cweb_response["Address"]["Value"]["Line1"]
+            self.cweb_zip = cweb_response["Address"]["Value"]["PostalCode"]
+            self.cweb_city = cweb_response["Address"]["Value"]["City"]
+            self.cweb_country = self._get_country(
+                cweb_response["Address"]["Value"]["CountryCode"]
+            )
+        elif self.cweb_address_enable and not cweb_has_address_values:
+            self.cweb_address_enable = False
+        else:
+            self.cweb_street = None
+            self.cweb_zip = None
+            self.cweb_city = None
+            self.cweb_country = None
+
+    def _populate_balans_cweb(self, cweb_response):
+        self.cweb_balance_data_enable = cweb_response["Balances"]["IsEnabled"]
+        cweb_has_balance_values = cweb_response["Balances"]["Value"]
+        if self.cweb_balance_data_enable and cweb_has_balance_values:
+            currency = self.env["res.currency"].search([("name", "=", "EUR")])
+            self.cweb_currency_id = currency
+            self.cweb_balance_year = cweb_response["Balances"]["Value"]["Balans"][0][
+                "BookYear"
+            ]
+            balans_data = cweb_response["Balances"]["Value"]["Balans"][0]["BalansData"][
+                "BalansData"
+            ]
+            for data in balans_data:
+                if data["Key"] == "CloseDate":
+                    try:
+                        self.cweb_closed_date = datetime.strptime(
+                            str(data["Value"]), "%Y-%m-%d"
+                        )
+                    except ValueError:
+                        self.cweb_closed_date = None
+                if data["Key"] == "Rub10_15":
+                    value = data["Value"]
+                    if value or value == 0:
```

Comment:
Actually, what is the data type of these values in the cweb soap service ? If it is string, then it will never be `0`.

Reply:
All thoses tests has been set in place because we can have
'Value': None
'Value': '-14397'
'Limit': Decimal('0')
I don't know if all the fields has this but i placed this security for all monetary fields -> you want me to remove it or ask the customer how is a balance "0" represented in the call's result ?